src/main/webapp/authorization.js

```diff
@@ -1,56 +1,115 @@
 // GCP Generated Github OAuth configuration
-const config = {
+const firebaseConfig = {
   apiKey: 'AIzaSyAR88Giah8cCEAvT_zDSIREWvgIIAeS8yY',
   authDomain: 'step2020-279820.firebaseapp.com',
 };
 
 /**
- * Initialize firebase if firebase is not already initialized. This method can
- * be called as many times as we need without any side effects.
+ * Object used for Github authorization and as a wrapper for firebase.
+ * It handles Firebase initialization on instantiation and holds nifty
+ * functions for handling github authorization. We are using a class to have
+ * access tokens persist for ease opf use on the backend
+ *
+ * Make sure to include the scripts below in any HTML File that uses this class
+ *
+ * <script src="https://www.gstatic.com/firebasejs/7.15.5/firebase.js"></script>
+ * <script type="module" src="authorization.js"></script>
+ *
+ * NOTE: The authorization.js file location might change.
  */
-function initializeFirebase() {
-  const firebase = getFirebase();
-  if (firebase.apps.length === 0) {
-    firebase.initializeApp(config);
+class GithubAuthorizer {
+  constructor() { // eslint-disable-line
```

Comment:
Q: Is there a reason why you disabled eslint on this line?

Reply:
Yeah, Eslint was angry about the constructor needing a JSDoc but was also angry about the JSDoc

Reply:
it was very confusing

Reply:
Suggested a fix!